static/app/views/settings/organizationTeams/teamMembersRow.tsx

```diff
@@ -0,0 +1,146 @@
+import styled from '@emotion/styled';
+
+import Button from 'sentry/components/button';
+import IdBadge from 'sentry/components/idBadge';
+import {PanelItem} from 'sentry/components/panels';
+import RoleSelectControl from 'sentry/components/roleSelectControl';
+import {IconSubtract} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Member, Organization, User} from 'sentry/types';
+import {
+  hasOrgRoleOverwrite,
+  RoleOverwriteIcon,
+} from 'sentry/views/settings/organizationTeams/roleOverwriteWarning';
+
+const TeamMembersRow = (props: {
+  hasWriteAccess: boolean;
+  member: Member;
+  organization: Organization;
+  removeMember: (member: Member) => void;
+  updateMemberRole: (member: Member, newRole: string) => void;
+  user: User;
+}) => {
+  const {organization, member} = props;
+
+  return (
+    <TeamRolesPanelItem key={member.id}>
+      <div>
+        <IdBadge avatarSize={36} member={member} useLink orgId={organization.slug} />
+      </div>
+      <div>
+        <TeamRoleSelect {...props} />
```

Comment:
personally i really dislike prop drilling like this.

Can we be more verbose about this and only pass specified props as needed?